src/uct/tcp/tcp_cm.c

```diff
@@ -416,12 +416,25 @@ void uct_tcp_cm_remove_ep(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep)
     ep->flags &= ~UCT_TCP_EP_FLAG_ON_MATCH_CTX;
 }
 
+static const char*
+uct_tcp_cm_format_addrs_pair(char *buffer, size_t max_length, const char *format,
```

Comment:
What about another approach:
```
#define SOCKADDR_TO_STR(sock_addr) \
    ({ \
        char buf[UCS_SOCKADDR_STRING_LEN + 1]; \
        ucs_sockaddr_str(sock_addr, buf, UCS_SOCKADDR_STRING_LEN); \
        buf; \
    })

ucs_assertv_always(..., "ucs_sockaddr_cmp(%s, %s) failed",
                    SOCKADDR_TO_STR(s1), SOCKADDR_TO_STR(s2))
```

Reply:
No, this does not work, need to allocate with alloca instead:
```
#define SOCKADDR_TO_STR(sock_addr) \
    ({ \
        char *buf = ucs_alloca(UCS_SOCKADDR_STRING_LEN); \
        ucs_sockaddr_str(sock_addr, buf, UCS_SOCKADDR_STRING_LEN); \
        buf; \
    })

Reply:
Wait, now it might not be safe to call `SOCKADDR_TO_STR` in `ucs_assertv_always`, because of:
> Do not use alloca inside the arguments of a function call
https://www.gnu.org/software/libc/manual/html_node/Variable-Size-Automatic.html#index-alloca-function

So the code would be:
```
s1_str = SOCKADDR_TO_STR(s1);
s2_str = SOCKADDR_TO_STR(s2);
ucs_assertv_always(..., "ucs_sockaddr_cmp(%s, %s) failed", s1_str, s2_str)

Reply:
Won't `buf` be freed after its scope of visibility, i.e.  `}` inside of macro?

Reply:
That's a compound expression, should be safe wrt alloca. The entire ucs_array is based on this idea:
```
#define UCS_ARRAY_ALLOC_ONSTACK(_array_type, _capacity) \
    ({ \
        typedef ucs_typeof(*((_array_type*)NULL)->buffer) value_t; \
        (value_t*)ucs_alloca((_capacity) * sizeof(value_t)); \
    })

Reply:
Ok, just variable allocation on stack looks safe vs a macro which is expected to be used like a function, but there is a limitation that it can't be used as an argument of other function

Reply:
maybe pass ep as parameter instead of s1,s2?
`static const char *uct_tcp_cm_ep_addrs_str(uct_tcp_ep_t *ep, char *buffer, size_t max_length)`